compiler/ilgen/MethodBuilder.hpp

```diff
@@ -73,7 +74,7 @@ class MethodBuilder : public TR::IlBuilder
 
    TR::TypeDictionary *typeDictionary()                      { return _types; }
 
-   const char *getDefiningFile()                             { return _definingFile; }
+   std::string getDefiningFile()                             { return _definingFile; }
```

Comment:
Allocation of memory is a subject requiring care in the compiler component, please see answer below.